### DISCUSSION TITLE
Add Data base backup /export

### DISCUSSION THREAD

**Discussion by ZayrixTheAvali** [Paymenter/Paymenter#383] [Ideas]
What this would be able to do is maybe have a section were people have a section to have there database export or backup in some type of way. Either externaly or were they can download it to there local machine.

**Comment by Poseidon281** [comment]
This is a .sh file for a database backup script with Discord functions. I made the echo in Dutch but that does not effect anything

#!/bin/bash

backup_dir="" # Set this to the directory you want to back it up to
log_file="backup_log_$(date +'%Y-%m-%d_%H-%M-%S').log" # Set this to the directory and file you want your logs in
remote_server="" # Set your server IP here can be external server
remote_mysql_user="" # Set your server username here 
remote_mysql_password="" # Set your server password here 
WEBHOOK_URL="https://discord.com/api/webhooks" # Set this to your Discord webhook

current_date=$(date +"%d-%m-%Y_%H-%M-%S")

databases=$(mysql -h "$remote_server" -u "$remote_mysql_user" -p"$remote_mysql_password" -e "SHOW DATABASES;" | grep -Ev "(Database|information_schema|performance_schema|mysql)")

touch "$log_file"

for db in $databases; do
    echo "Backup aan het maken voor: $db" | tee -a "$log_file"
    mysqldump -h "$remote_server" -u "$remote_mysql_user" -p"$remote_mysql_password" "$db" > "$backup_dir/$db-$current_date.sql" 2>> "$log_file"
    curl -X POST -H "Content-Type: multipart/form-data" -F "file=@$backup_dir/$db-$current_date.sql" "$WEBHOOK_URL" >> "$log_file" 2>&1
done

echo "Backups zijn successvol gemaakt en in $backup_dir gestopt. Ook zijn alle bestanden naar Discord gestuurd met $WEBHOOK_URL ." | tee -a "$log_file"
